add lua data type entry (#2917)
Codecademy/docs: content/lua/concepts/data-types/data-types.md

---
Title: 'Data Type' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
Description: 'In Lua, there are many **Data Types**. Commonly used data types are: Number, Boolean, String, and Nil.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
 - 'Computer Science'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
  - 'Booleans'
  - 'Data Types'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first

>   - 'learn-lua'

Reviewer: ```suggestion
  - 'learn-lua'
  - 'paths/computer-science'
```

---

> In Lua, there are many **Data Types**. Commonly used data types are:

Reviewer: ```suggestion
In Lua, there are many **Data Types**. Some commonly used data types are:
```

* Number
* Boolean
* String
* Nil 

## type()
Lua has a built-in `type()` function to retrieve the data type.
```lua
location = 'Maple Road'

> print(type(location)) -- This outputs string

Author: Ohh I see what you mean! I think it makes sense for that change for clarity.

> print(type(location)) -- This outputs string

Author: ```suggestion
print(type(location)) -- This outputs "string"
```

---
Title: 'Data Type' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed

> Description: 'In Lua, there are many **Data Types**. Commonly used data types are: Number, Boolean, String, and Nil.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)

Reviewer: ```suggestion
Description: 'In Lua, there are many Data Types. Commonly used data types are: Number, Boolean, String, and Nil.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
```

Missed this, we can't have styling in the metadata description

Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
 - 'Computer Science'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
  - 'Booleans'
  - 'Data Types'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'learn-lua'
  - 'paths/computer-science'
---
In Lua, there are many **Data Types**. Some commonly used data types are:
* Number
* Boolean
* String
* Nil 

> ## type()

Author: ```suggestion
## type()

```

Lua has a built-in `type()` function to retrieve the data type.
```lua
location = 'Maple Road'
print(type(location)) -- This outputs "string"

> print(type(12/3)) -- This outputs number

Reviewer: for consistency

```

> ## Type Conversion

Author: ```suggestion
## Type Conversion

```

---

> Title: 'Data Type' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed

Reviewer 2: ```suggestion
Title: 'Data Types' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
```

> Description: 'In Lua, there are many Data Types. Commonly used data types are: Number, Boolean, String, and Nil.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)

Reviewer 2: ```suggestion
Description: 'Data types govern the range of values and functionality of those objects in a language.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
```

Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
 - 'Computer Science'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
  - 'Booleans'
  - 'Data Types'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'learn-lua'
  - 'paths/computer-science'
---

> In Lua, there are many **Data Types**. Some commonly used data types are:

Reviewer 2: ```suggestion

Lua, like Python, is a dynamically typed language, which means the the type does not have to be explicitly declared with the variable. Lua has eight basic **data types**,  some commonly used data types are:
```

> * Number

Reviewer 2: ```suggestion
* Number: Represents double-percision floating point values (there are is no `int` type in Lua). 
```

Reviewer 2: Boolean and string are fairly straightforward but I think number and nil are less common. It might be good to briefly define each of these as above

Author: that makes sense thanks you

* Boolean
* String
* Nil 

## type()

> Lua has a built-in `type()` function to retrieve the data type.

Reviewer 2: ```suggestion
Lua has a built-in `type()` function to retrieve the data type.

```

```lua
location = 'Maple Road'
print(type(location)) -- This outputs "string"

print(type(12/3)) -- This outputs "number"

> ```

Reviewer 2: add an empty line after the code block

> ```

Author: ```suggestion
```

```

## Type Conversion

> In Lua, **type coercion** automatically happens when strings and numbers are concatenated together or when a string with a numeric value appears in an arithmetic expression.

Reviewer 2: add an empty line after this sentence

```lua
--  The first type coercion is used in a concatenation to change a number to a string.

temperature = 56

print('Today’s temperature is ' .. temperature .. ' degrees.') -- This prints “Today’s temperature is 56 degrees.”

-- The second type of coercion is used during arithmetic expressions.
print('53' + 7)

-- This outputs 60 because it converted “53” to a number for the arithmetic.
```

> Types can be manually converted by using:
> 
> * `tostring()` function to convert any data type to a string type 

Reviewer 2: ```suggestion
Types can be manually converted by using the `tostring()` function to convert any data type to a string type:
```

```lua
player1Pts = 55

print('Player 1 has ' .. tostring(player1Pts) .. ' points')
-- This prints “Player 1 has 55 points”
```

> * `tonumber()` function to convert any data type to a number type

Reviewer 2: ```suggestion
The `tonumber()` function is used to convert any data type to a number type:
```
